pkg/crinit/options/options.go

```diff
@@ -0,0 +1,391 @@
+	flags.Int32Var(&o.ApiServerNodePortPort, apiserverPortFlag, 0,
+		"Preferred port to use for the cluster registry API server NodePort service. Set to 0 to randomly assign a port. Valid only if '"+APIServerServiceTypeFlag+"=NodePort'.")
+}
+
+// SetName sets the name of the cluster registry.
+func (o *SubcommandOptions) SetName(args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("NAME is required")
+	}
+	o.Name = args[0]
+	return nil
+}
+
+// ValidateCommonOptions validates the options that are shared across
+// different sub-commands.
+func (o *SubcommandOptions) ValidateCommonOptions() error {
+	serverName = fmt.Sprintf("%s-%s", o.Name, APIServerNameSuffix)
+	serverCredName = fmt.Sprintf("%s-%s", serverName, CredentialSuffix)
+
+	if o.ApiServerServiceType != v1.ServiceTypeLoadBalancer &&
+		o.ApiServerServiceType != v1.ServiceTypeNodePort {
+		return fmt.Errorf("invalid %s: %s, should be either %s or %s",
+			APIServerServiceTypeFlag, o.ApiServerServiceType,
+			v1.ServiceTypeLoadBalancer, v1.ServiceTypeNodePort)
+	}
+
+	if o.ApiServerAdvertiseAddress != "" {
+		ip := net.ParseIP(o.ApiServerAdvertiseAddress)
+		if ip == nil {
+			return fmt.Errorf("invalid %s: %s, should be a valid ip address",
+				apiserverAdvertiseAddressFlag, o.ApiServerAdvertiseAddress)
+		}
+		if o.ApiServerServiceType != v1.ServiceTypeNodePort {
+			return fmt.Errorf("%s should be passed only with '%s=NodePort'",
+				apiserverAdvertiseAddressFlag, APIServerServiceTypeFlag)
+		}
+	}
+
+	if o.ApiServerNodePortPort != 0 {
+		if o.ApiServerServiceType != v1.ServiceTypeNodePort {
+			return fmt.Errorf("%s should be passed only with '%s=NodePort'",
+				apiserverPortFlag, APIServerServiceTypeFlag)
+		}
+		o.ApiServerNodePortPortPtr = &o.ApiServerNodePortPort
+	} else {
+		o.ApiServerNodePortPortPtr = nil
+	}
+
+	if o.ApiServerNodePortPort < 0 || o.ApiServerNodePortPort > 65535 {
+		return fmt.Errorf("Please provide a valid port number for %s", apiserverPortFlag)
+	}
+
+	return nil
+}
+
+// marshalOptions marshals options if necessary.
+func (o *SubcommandOptions) MarshalOptions() error {
+	if o.ApiServerOverridesString == "" {
+		return nil
+	}
+
+	argsMap := make(map[string]string)
+	overrideArgs := strings.Split(o.ApiServerOverridesString, ",")
+	for _, overrideArg := range overrideArgs {
+		splitArg := strings.SplitN(overrideArg, "=", 2)
+		if len(splitArg) != 2 {
+			return fmt.Errorf("wrong format for override arg: %s", overrideArg)
+		}
+		key := strings.TrimSpace(splitArg[0])
+		val := strings.TrimSpace(splitArg[1])
+		if len(key) == 0 {
+			return fmt.Errorf("wrong format for override arg: %s, arg name cannot be empty", overrideArg)
+		}
+		argsMap[key] = val
+	}
+
+	o.ApiServerOverrides = argsMap
+
+	return nil
+}
+
+// CreateNamespace creates the cluster registry namespace.
+func (o *SubcommandOptions) CreateNamespace(cmdOut io.Writer,
```

Comment:
I think (perhaps in a follow up) this code should be moved to the common package, and should take `SubcommandOptions` as a parameter rather than having it be the receiver.